src/main/kotlin/traffic_simulation/Vehicle.kt

```diff
@@ -27,6 +28,10 @@ class Vehicle(val id: Int, val wannaDriveInHours: MutableList<Int>) {
         if (!wannaDriveInHours.contains(timestep)) {
             delay -= 1
         }
+
+        if (delay <= 0 && wannaDriveInHours.contains(timestep)){
```

Comment:
Genau, `delayed` enthält dann nur Stunden, in denen das Fahrzeug wegen Stau nicht fahren konnte. Wenn ein Fahrzeug wegen Staus in der Vergangenheit "hinterherhecheln" muss, ist das daran nicht direkt abzulesen. Allerdings ist ja für jedes Auto bekannt, wann es fahren _wollte_, wann es gefahren _ist_ und wann es _nicht_ fahren _konnte_. Daraus ließe sich alles weitere ableiten.